src/app/Main/Body/QueryPage/QueryPageServices/TableServices/generateOneTableRowObj.service.ts

```diff
@@ -13,29 +13,45 @@ export function generateOneTableRowObjService(
 ): Record<string, string> {
     const tempStatTable: Record<string, string> = {};
     tempStatTable.name = rowName;
-    columnNameValues.forEach((colName: string | number) => {
-        if (
+    columnNameValues.forEach((colName: string) => {
+        const isolateGroupIncudesRowAndCol: boolean =
             _.includes(uniqIsolateColValues, colName) &&
-            _.includes(uniqIsolateRowValues, rowName)
-        ) {
+            _.includes(uniqIsolateRowValues, rowName);
+
+        if (isolateGroupIncudesRowAndCol) {
+            let breakIsolateCountLoop = false;
             isolateCountGroups.forEach((isolateGroup) => {
-                if (
-                    (isolateGroup[rowAttribute] === rowName ||
-                        isolateGroup[rowAttribute] === undefined) &&
-                    (isolateGroup[colAttribute] === colName ||
-                        isolateGroup[colAttribute] === undefined)
-                ) {
-                    const statTableKey =
-                        isolateGroup[colAttribute] === undefined
-                            ? allValuesText
-                            : isolateGroup[colAttribute];
-                    tempStatTable[statTableKey] = String(isolateGroup.count);
+                if (breakIsolateCountLoop) {
+                    return;
+                }
+
+                const isolateGroupRowValue = isolateGroup[rowAttribute];
+                const isolateGroupColValue = isolateGroup[colAttribute];
+                const isolateCount = isolateGroup.count;
+
+                const onlyRowNameIsInIsolateGroup: boolean =

```

Comment:
Reject: The algorithm is still problematic: E.g. if `ig[row] === rowName && ig[col] === undefined` comes first in the isolateGroups the loop ends always in setting the allValues entry and the other cols will be ignored. So a solution would be to first scan through the isolateGroups in the loop and set booleans for things that where found during the loop. After the loop you can then check these booleans and decide what to do.
As stated it would be good to convert the isolateCount data from the API to a more convenient format  before doing anything with it. I would create a new ticket for that and if you like we can develop a conversion  algorithm together.

Reply:
No, the other columns are not ignored. The col-for-each loop is independent of the boolean break. The break is only for the ig iteration that starts for each column. 
But yes, of course we can rethink this alogrithm in a later ticket.

Reply:
Re-Comment (this comes first) (messed up with github review, sorry): Yes, but for each col-for-each the ig loop will success on the row = rowName and col = undefined check and set the allValuesText in my example situtation. Then the ig loop cancelling boolean is set and all other igs are ignored.

Reply:
According to api, there is only one associated isolate group per field. That means, when I have found it, I can cancel the loop over the ig, because I won't find any in the following ones anyway. Sure it depends on the API, but I don't think it's that buggy.

Reply:
Obsolete, due to misunderstanding, the api response has a different semantic for the discussed problematic cases.